src/databricks/labs/remorph/reconcile/connectors/snowflake.py

```diff
@@ -1,28 +1,45 @@
 import re
 
-from pyspark.errors import PySparkException
-from pyspark.sql import DataFrame
+from pyspark.sql import DataFrame, SparkSession
 from pyspark.sql.functions import col
+from sqlglot import Dialects
 
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.connectors.jdbc_reader import JDBCReaderMixin
+from databricks.labs.remorph.reconcile.connectors.secrets import SecretsMixin
 from databricks.labs.remorph.reconcile.constants import SourceDriver, SourceType
 from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Schema
+from databricks.sdk import WorkspaceClient
 
 
-class SnowflakeDataSource(DataSource):
+class SnowflakeDataSource(DataSource, SecretsMixin, JDBCReaderMixin):
+
+    def __init__(
+        self,
+        engine: Dialects,
+        spark: SparkSession,
+        ws: WorkspaceClient,
+        secret_scope: str,
+    ):
+        self._engine = engine
+        self._spark = spark
+        self._ws = ws
+        self._secret_scope = secret_scope
+
     @property
     def get_jdbc_url(self) -> str:
         return (
-            f"jdbc:{SourceType.SNOWFLAKE.value}://{self._get_secrets('account')}.snowflakecomputing.com"
-            f"/?user={self._get_secrets('sfUser')}&password={self._get_secrets('sfPassword')}"
-            f"&db={self._get_secrets('sfDatabase')}&schema={self._get_secrets('sfSchema')}"
-            f"&warehouse={self._get_secrets('sfWarehouse')}&role={self._get_secrets('sfRole')}"
+            f"jdbc:{SourceType.SNOWFLAKE.value}://{self._get_secret('sfAccount')}.snowflakecomputing.com"
+            f"/?user={self._get_secret('sfUser')}&password={self._get_secret('sfPassword')}"
+            f"&db={self._get_secret('sfDatabase')}&schema={self._get_secret('sfSchema')}"
+            f"&warehouse={self._get_secret('sfWarehouse')}&role={self._get_secret('sfRole')}"
         )
 
-    def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOptions) -> DataFrame:
+    def read_data(
+        self, catalog: str, schema: str, table: str, query: str, options: JdbcReaderOptions | None
```

Comment:
Just for lateral reference: 
I'm anticipating `mypy` will throw an error for this `JdbcReaderOptions | None`.

Reply:
we will fix this then.